File: задание 1/main.rb
print "Input length: "
N = gets.chomp.to_i

print "Input width: "
M = gets.chomp.to_i

print "Input all commands: "
commands = gets.chomp

x = 0
y = 0

commands.each_char do |command|
    case command
    when 'L'
        x -= 1
    when 'R'
        x += 1
    when 'D'
        y += 1
    when 'U'
        y -= 1
    else
        raise ArgumentError, "Invalid command - #{command}"
    end

    if y.abs >= N || x.abs >= M
        puts "No"
        exit 1
    end
end

puts "(#{M - x.abs}, #{N - y.abs})"

File: задание 1/main.go
package main

import (
	"fmt"
	"os"
)

func main() {
	var N, M int
	fmt.Print("Input length: ")
	_, err := fmt.Scan(&N)
	if err != nil {
		fmt.Println("Error:", err)
		os.Exit(1)
	}

	fmt.Print("Input width: ")
	_, err = fmt.Scan(&M)
	if err != nil {
		fmt.Println("Error:", err)
		os.Exit(1)
	}

	var commands string
	fmt.Print("Input all commands: ")
	_, err = fmt.Scan(&commands)
	if err != nil {
		fmt.Println("Error:", err)
		os.Exit(1)
	}

	x, y := 0, 0

	for _, command := range commands {
		switch command {
		case 'L':
			x -= 1
		case 'R':
			x += 1
		case 'D':
			y += 1
		case 'U':
			y -= 1
		default:
			fmt.Printf("Invalid command - %c\n", command)
			os.Exit(1)
		}

		if abs(y) >= N || abs(x) >= M {
			fmt.Println("No")
			os.Exit(0)
		}
	}

	fmt.Printf("(%d,%d)\n", M-abs(x), N-abs(y))
}

func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}
